### ISSUE TITLE
SPO: How to configure default Locale (date- and timeformats etc)

### ISSUE THREAD

**Issue by salbeck-sit** [microsoft/Microsoft365DSC#5531]
The current SPO-resources don't seem to include anything to configure regional settings like date- and time-formats.
I'm tempted to take this on but I am unsure of where to start. Graph is notoriously lacking but PnP seem to have something (see https://www.sharepointdiary.com/2019/06/sharepoint-online-change-regional-settings-using-powershell.html).
Then again, if this isn't done at the beginning, multiple sites may exist and need changing as well - this could be optional.

Anyone tried this ?

**Comment by FabienTschanz** [comment]
@salbeck-sit Nope, didn't try it. Do you want to add the this as a new resource or to an existing one? I think the code would be straightforward.

**Comment by salbeck-sit** [comment]
I think I already have it figured out. A SPO-colleague did something clever with SPOSiteScript and SPOSiteDesign.
The below example (the content of the site-script) configures timezone and locale (24-hour CET and danish in the example).

SPOSiteScript 'RegionalSettings'
        {
            Title        = "RegionalSettings"
            Content      = '{
                "$schema": "schema.json",
                "actions": [
                {
                    "verb": "setRegionalSettings",
                    "timeZone": 3,
                    "locale": 1030,
                    "sortOrder": 9,
                    "hourFormat": "24"
                }
                ],
                "bindata": {},
                "version": 1
            }'
            Description  = "RegionalSettings"
            Ensure       = "Present"
            ApplicationId         = $ConfigurationData.NonNodeData.ApplicationId;
            CertificateThumbprint = $ConfigurationData.NonNodeData.CertificateThumbprint;
            TenantId              = $OrganizationName;
            #Credential  = $Credscredential
        }
Followed by a SPOSIteDesign per site-type/webtemplate that includes the sitescript to ensure it runs for every site.

**Comment by FabienTschanz** [comment]
Gotcha. Then I'll close the issue. Thank you.
